File: 199-binary-tree-right-side-view/199-binary-tree-right-side-view.cpp

class Solution {
public:
    vector<int> rightSideView(TreeNode* root) {
        // vector<int> res;
        // helper(root,0,res);
        // return res;
        
         vector<int> res;
        if(!root) return res;
        
        queue<TreeNode*> q;
        q.push(root);
        
         while(!q.empty()){            
            int size = q.size();
            res.push_back(q.front()->val);
            
            for(int itr = 0; itr < size; itr++){
                TreeNode* temp = q.front();
                q.pop();                
                                
                if(temp->right) q.push(temp->right);
                if(temp->left) q.push(temp->left);
            }
        }
        return res;
    }
    
//     void helper(TreeNode *root,int level,vector<int> &res)
//     {
//         if(root==NULL) return;
//         if(res.size()==level) res.push_back(root->val);
//         helper(root->right,level+1,res);
//         helper(root->left,level+1,res);
            
//     }
};

File: 199-binary-tree-right-side-view/199-binary-tree-right-side-view.java

class Solution {
    public List<Integer> rightSideView(TreeNode root) {
        List<Integer> res=new ArrayList<>();
        helper(root,0,res);
        return res;
    }
    public void helper(TreeNode root,int level,List<Integer> res)
    {
        if(root==null) return;
        if(res.size()==level) res.add(root.val);
      
        
          helper(root.right,level+1,res);helper(root.left,level+1,res);
       
    }
}